Issue #3. Setup for Apache + PHP 7

johnrosswvsu opened the pull request:
Updated Read Me File for Apache + PHP 7 setup.

Updated docker-composer.yml file to use Apache.

Removed Nginx setup and add Apache set-up.

johnrosswvsu committed:
Issue #3. Update README.md file.

Corrected file to use Apache in documentation.

johnrosswvsu committed:
Issue #3. Update docker-composer.yml file.

Corrected YML configuration to use Apache + PHP set-up.

johnrosswvsu committed:
Issue #3. Remove Nginx set-up.

Removed Nginx setup from docker.

johnrosswvsu committed:
Issue #3. Add apache set-up.

Added vhost configuration and other bash script for setting up
Apache configurations.

Luukyb commented on README.md:
```diff
@@ -2,13 +2,13 @@
+### Apache + PHP
 
-This branch is still on progress. For more stable, use the nginx branches.
+This branch is still on progress. For more stable, use the apache branches.
```
Please delete this line.

johnrosswvsu replied:
Does this mean I have to remove this line also:


`
### PHP 7 or PHP 5?
You can switch between the latest release of PHP 7 or the latest of PHP 5, by
changing the first line of the file "docker/php/Dockerfile".
```
FROM php:5-fpm`

Luukyb reviewed (commented):
The Dockerfile seems wrong. It still uses php:7-fpm.
It should be using "FROM php:7-apache".

You also need RUN a2enmod rewrite to enable the rewrite module. RUN a2ensite 000-default to enable the host.

When testing on your local can you try to remove the existing containers for the folder "docker-compose rm", then "docker-compose up -d --force-recreate", then just to make sure "docker-compose build --no-cache --".

When opening your browser on Expresso PHP, you should see the PHP info and mention of PHP 7 running on Apache.
Thanks!

johnrosswvsu committed:
Issue #3. Totally change Dockerfile.

To use Apache instead of PHP FPM.

Luukyb commented on docker/php/Dockerfile:
```diff
@@ -16,7 +20,7 @@ RUN { \
         echo 'opcache.interned_strings_buffer=8'; \
         echo 'opcache.max_accelerated_files=4000'; \
-        echo 'opcache.revalidate_freq=0'; \
+        echo 'opcache.revalidate_freq=60'; \
```
Can we keep zero? This parameter to 60 is great in production, but not in local. You would not see your changes immediately.

johnrosswvsu replied:
Updated.

johnrosswvsu committed:
Issue #3. Update revalidate_freq default configuration for Apache.

Local works well with 0 rather than the 60 value, which is good
for live sites.
